cmd/maya-apiserver/spc-watcher/select_disk.go

```diff
@@ -137,28 +144,45 @@ func diskSelector(nodeDiskMap map[string]*diskList, poolType string) *nodeDisk {
 		},
 	}
 
-	// requiredDiskCount will hold the required number of disk that should be selected from a qualified
+	// diskCount will hold the number of disk that will be selected from a qualified
 	// node for specific pool type
-	var requiredDiskCount int
-	// If pool type is striped, 1 disk should be selected
+	var diskCount int
+	// minRequiredDiskCount will hold the required number of disk that should be selected from a qualified
+	// node for specific pool type
+	var minRequiredDiskCount int
+	// If pool type is striped, at least 1 disk should be selected
 	if poolType == string(v1alpha1.PoolTypeStripedCPV) {
-		requiredDiskCount = int(v1alpha1.StripedDiskCountCPV)
+		minRequiredDiskCount = int(v1alpha1.StripedDiskCountCPV)
 	}
-	// If pool type is mirrored, 2 disks should be selected
+	// If pool type is mirrored, at least 2 disks should be selected
 	if poolType == string(v1alpha1.PoolTypeMirroredCPV) {
```

Comment:
Should we return error for anything other than 'Striped' or 'mirrored'?
And, how about a function that returns error or the minimum number of disks required based on the type passed to it?

Reply:
ok.. this just getting the constant value, so, new function won't be required. But, atleast, should return error from here if other than striped or mirrored.

Reply:
sorry for being late here.. How about code like this:
```--- a/cmd/maya-apiserver/spc-watcher/select_disk.go
+++ b/cmd/maya-apiserver/spc-watcher/select_disk.go
@@ -140,22 +140,29 @@ func diskSelector(nodeDiskMap map[string]*diskList, poolType string) *nodeDisk {
        // requiredDiskCount will hold the required number of disk that should be selected from a qualified
        // node for specific pool type
        var requiredDiskCount int
-       // If pool type is striped, 1 disk should be selected
-       if poolType == string(v1alpha1.PoolTypeStripedCPV) {
-               requiredDiskCount = int(v1alpha1.StripedDiskCountCPV)
-       }
-       // If pool type is mirrored, 2 disks should be selected
-       if poolType == string(v1alpha1.PoolTypeMirroredCPV) {
-               requiredDiskCount = int(v1alpha1.MirroredDiskCountCPV)
-       }
+
+       // Lets have a check in nodeDiskAlloter for valid poolType string
+
+       // Get the minimum required number of disks based on poolType
+       minRequiredDiskCount = v1alpha1.MinDiskCountBasedOnPoolType[poolType]
+
        // Range over the nodeDiskMap map to get the list of disks
        for node, val := range nodeDiskMap {
 
                // If the current disk count on the node is less than the required disks
                // then this is a dirty node and it will not qualify.
-               if len(val.items) < requiredDiskCount {
+               if len(val.items) < minRequiredDiskCount {
+                       logrus.Infof("Mininum number of disks is not available for node %s.. hence, skipping")
                        continue
                }
+
+               requiredDiskCount = minRequiredDiskCount;
+
+               // If provisiongType is Manual, consume all the disks
+               if provisiongType == ProvisioningTypeManual {
+                       requiredDiskCount = (len(val.items) / minRequiredDiskCount ) * minRequiredDiskCount;
+               }
+
                // Select the required disk from qualified nodes.
                for i := 0; i < requiredDiskCount; i++ {
                        selectedDisk.disks.items = append(selectedDisk.disks.items, val.items[i])
```

Reply:
above needs a const map MinDiskCountBasedOnPoolType for each valid poolType